test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Roundtrip/JsonReaderWriterInjectionTests.cs

```diff
@@ -420,99 +408,126 @@ public void WriteName(string name)
 
-            public void WriteValue(Guid value)
-            {
-                throw new NotImplementedException();
-            }
+            public void WriteValue(Guid value) => throw new NotImplementedException();
 
-            public void WriteValue(decimal value)
-            {
-                throw new NotImplementedException();
-            }
+            public void WriteValue(decimal value) => throw new NotImplementedException();
 
-            public void WriteValue(DateTimeOffset value)
-            {
-                throw new NotImplementedException();
-            }
+            public void WriteValue(DateTimeOffset value) => throw new NotImplementedException();
 
-            public void WriteValue(TimeSpan value)
-            {
-                throw new NotImplementedException();
-            }
+            public void WriteValue(TimeSpan value) => throw new NotImplementedException();
+
+            public void WriteValue(byte value) => throw new NotImplementedException();
+
+            public void WriteValue(sbyte value) => throw new NotImplementedException();
 
-            public void WriteValue(byte value)
+            public void WriteValue(string value)
             {
-                throw new NotImplementedException();
+                this.textWriter.Write("\"{0}\",", value);
             }
 
-            public void WriteValue(sbyte value)
+            public void WriteValue(byte[] value) => throw new NotImplementedException();
+
+            public void WriteValue(Date value) => throw new NotImplementedException();
+
+            public void WriteValue(TimeOfDay value) => throw new NotImplementedException();
+
+            public void WriteRawValue(string rawValue) => throw new NotImplementedException();
+
+            public void Flush()
             {
-                throw new NotImplementedException();
+                this.textWriter.Flush();
             }
 
-            public void WriteValue(string value)
+            public Task StartPaddingFunctionScopeAsync() => throw new NotImplementedException();
+
+            public Task EndPaddingFunctionScopeAsync() => throw new NotImplementedException();
+
+            public async Task StartObjectScopeAsync()
             {
-                this.textWriter.Write("\"{0}\",", value);
+                await this.textWriter.WriteAsync('<');
             }
 
-            public void WriteValue(byte[] value)
+            public async Task EndObjectScopeAsync()
             {
-                throw new NotImplementedException();
+                await this.textWriter.WriteAsync('>');
             }
 
-            public void WriteValue(Date value)
+            public Task StartArrayScopeAsync() => throw new NotImplementedException();
+
+            public Task EndArrayScopeAsync() => throw new NotImplementedException();
+
+            public async Task WriteNameAsync(string name)

```

Comment:
Does this need to be async? It's almost impossible that the code in the `if` block would throw so the last line in the function could change to `return this.textWriter.WriteAsync(string.Format("\"{0}\":", name));`

Reply:
It would through if `name` is `null`.